ai21/clients/studio/resources/chat/base_chat_completions.py

```diff
@@ -52,6 +58,11 @@ def _create_body(
                 "stop": stop,
                 "n": n,
                 "stream": stream,
+                "tools": [_to_dict(tools) for tools in tools] if tools is not NOT_GIVEN else NOT_GIVEN,
+                "response_format": _to_dict(response_format) if response_format is not NOT_GIVEN else NOT_GIVEN,
+                "documents": (
+                    [_to_dict(document) for document in documents] if documents is not NOT_GIVEN else NOT_GIVEN
```

Comment:
Perhaps this logic should be moved to `remove_not_given` by checking the type of the parameter. wdyt?

Reply:
You mean to _to_dict? that can work. I.e. -> 

```
def _to_dict(model_object: BaseModel | NotGiven, **kwargs) -> Dict[str, Any] | NotGiven:
    if isinstance(model_object, NotGiven):
        return model_object
    if IS_PYDANTIC_V2:
        return model_object.model_dump(**kwargs)

    return model_object.dict(**kwargs)
```

Reply:
I think if you make these objects `TypedDict` you wouldn't need to do this check at all. They will be pass as dict without the need to serialize them. The call would look as simple as -

```python
  return remove_not_given(
            {
                "model": model,
                "messages": [_to_dict(message) for message in messages],
                "temperature": temperature,
                "max_tokens": max_tokens,
                "top_p": top_p,
                "stop": stop,
                "n": n,
                "stream": stream,
                "tools": tools,
                "response_format": response_format,
                "documents": documents,
                **kwargs,
            }
```